### PR TITLE
Only access variables in the state in gamma_law_general

### PR DESCRIPTION
**Pull request by maxpkatz**

### CODE REVIEW

**Comment by maxpkatz** [comment]
Easier to review without whitespace changes

File: EOS/gamma_law_general/actual_eos.H [hunk 1]
```diff
@@ -48,190 +48,245 @@ template <typename I, typename T>
...
-      // pressure, enthalpy and xmass are inputs
+          // pressure, enthalpy and xmass are inputs
 
-      // Solve for temperature and density
+          // Solve for temperature and density
 
-      state.rho = state.p / state.h * eos_gamma / (eos_gamma - 1.0);
-      state.T = state.p * state.mu * m_nucleon / (C::k_B * state.rho);
+          if constexpr (has_pressure<T>::value) {
+              state.rho = state.p / state.h * eos_gamma / (eos_gamma - 1.0);
```
Comment (zingale) [inline EOS/gamma_law_general/actual_eos.H:163]: needs both p and h

Comment (maxpkatz) [reply EOS/gamma_law_general/actual_eos.H:163]: Fixed

**Review by zingale — APPROVED** [review]

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Microphysics/test-suite/gfortran/2021-01-03-003/index.html

**Pull request merged**